Automattic/wp-calypso
# Themes: Move Components around (search, selection, ...)

_PR for #9671 (part 1/2):_
Moving `ThemesSearchCard` up at `ThemeShowcase` level, to allow a more flexible architecture and having search trigger multiple data containers.

No UI change. 
Pure refactoring.


### Testing

* Make sure everything in `/design` runs properly.
* Try searching a string.
* Try searching filters (i.e. `color:green`).
* Try changing All / Free / Premium.
* Try switching from single site to all sites.
* Try opening directly from a filtered/search complex URL
* Try logged out
